nixos/modules/services/x11/extra-layouts.nix

```diff
@@ -0,0 +1,154 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  layouts = config.services.xserver.extraLayouts;
+
+  layoutOpts = {
+    options = {
+      description = mkOption {
+        type = types.str;
+        description = "A short description of the layout.";
+      };
+
+      languages = mkOption {
+        type = types.listOf types.str;
+        description =
+        ''
+          A list of languages provided by the layout.
+          (Use ISO 639-2 codes, for example: "eng" for english)
+        '';
+      };
+
+      filepath = mkOption {
+        type = types.path;
+        description = "The path to the layout file.";
```

Comment:
Oh actually one more thing: you should add some info on the format of this file, maybe link to https://wiki.archlinux.org/index.php/X_KeyBoard_extension#Basic_examples

Reply:
I linked the official documentation in the description of `extraLayouts`. I'm adding this as well as it provides more examples.